src/main/java/org/javarosa/xpath/expr/XPathDistanceFunc.java

```diff
@@ -39,11 +40,15 @@ public static Double distance(Object from, Object to) {
             return new Double(-1.0);
         }
 
-        // Casting and uncasting seems strange but is consistent with the codebase
-        GeoPointData castedFrom = new GeoPointData().cast(new UncastData(unpackedFrom));
-        GeoPointData castedTo = new GeoPointData().cast(new UncastData(unpackedTo));
+        try {
+            // Casting and uncasting seems strange but is consistent with the codebase
+            GeoPointData castedFrom = new GeoPointData().cast(new UncastData(unpackedFrom));
+            GeoPointData castedTo = new GeoPointData().cast(new UncastData(unpackedTo));
 
-        return new Double(GeoPointUtils.computeDistanceBetween(castedFrom, castedTo));
+            return new Double(GeoPointUtils.computeDistanceBetween(castedFrom, castedTo));
+        } catch (NumberFormatException e) {
+            throw new XPathTypeMismatchException("distance() function requires arguments containing numeric values only");
```

Comment:
Might be useful to display the arguments in the error message?

Reply:
@wpride Good suggestion, added